plugins/woo-ai/src/image-background-removal/background-removal-link.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+import { __experimentalUseBackgroundRemoval as useBackgroundRemoval } from '@woocommerce/ai';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import MagicIcon from '../../assets/images/icons/magic.svg';
+import { FILENAME_APPEND } from './constants';
+import {
+	uploadImageToLibrary,
+	getCurrentAttachmentDetails,
+} from './image_utils';
+
+const getErrorMessage = ( errorCode?: string ) => {
+	switch ( errorCode ) {
+		case 'invalid_image_file':
+			return __( 'Invalid image', 'woocommerce' );
+		case 'image_file_too_small':
+			return __( 'Image too small', 'woocommerce' );
+		case 'image_file_too_large':
+			return __( 'Image too large', 'woocommerce' );
+		default:
+			return __( 'Something went wrong', 'woocommerce' );
+	}
+};
+
+export const BackgroundRemovalLink = () => {
+	const { fetchImage } = useBackgroundRemoval();
+	const [ state, setState ] = useState< '' | 'generating' | 'uploading' >(
+		''
+	);
+	const [ displayError, setDisplayError ] = useState< string | null >( null );
+
+	const onRemoveClick = async () => {
+		try {
+			setState( 'generating' );
+
+			const { url: imgUrl, filename: imgFilename } =
+				getCurrentAttachmentDetails();
+
+			if ( ! imgUrl ) {
+				setDisplayError( getErrorMessage() );
+				return;
+			}
+
+			const originalBlob = await fetch( imgUrl ).then( ( res ) =>
+				res.blob()
+			);
+
+			const bgRemoved = await fetchImage( {
+				imageFile: new File( [ originalBlob ], imgFilename ?? '', {
+					type: originalBlob.type,
+				} ),
+			} );
+
+			setState( 'uploading' );
+
+			await uploadImageToLibrary( {
+				imageBlob: bgRemoved,
+				libraryFilename: `${ imgFilename }${ FILENAME_APPEND }.${ bgRemoved.type
+					.split( '/' )
+					.pop() }`,
+			} );
+			setState( '' );
+		} catch ( err ) {
+			//eslint-disable-next-line no-console
+			console.error( err );
+			setDisplayError(
+				getErrorMessage( ( err as { code?: string } )?.code ?? '' )
+			);
+		}
+	};
+
+	if ( displayError ) {
+		return (
+			<span className="background-removal-error">{ displayError }</span>
```

Comment:
It'd be good to have a retry option here. Also, the text color is the same as the "Delete Permanently" link above; it might be slightly confusing.

<img width="163" alt="image" src="https://github.com/woocommerce/woocommerce/assets/73110514/1c68b422-4026-41ff-9b96-76989d3f66a4">

Reply:
Hmm, I did this on purpose, but now that you mention it I agree. I'll just remove the red color now. Do you think there's another color that's more appropriate? 

Done in e0b0e06a02dcfdc833dcdc193240a21701e2114d

Reply:
Maybe we can show a notice bar (not ideal, I know 😅) or a snack bar error that dismisses automatically. I feel anything we add there will seem out of place since those are all action buttons.

Reply:
I tried implementing the snackbar idea, although truthfully I don't really like it:

![image](https://github.com/woocommerce/woocommerce/assets/444632/94d2e6b3-c495-4515-90e1-2cb7d39d7f8a)


I just don't think the user would even notice the "notice." Do you mind if we keep this as is, and ping the rest of the team for suggestions that we can add in another issue? Since it's only Peter and I next week I'd like to get as much of this over the line as possible by EOD tomorrow.

Reply:
Sounds good. I still think having the error text right under the action buttons and removing the "Remove background" button without any retry option would be confusing. 

Sorry, I don't want to drag this out any longer, but how about keeping the "Remove background" button there (or replacing it with a retry button) and displaying a "notice" box under the button like this:

<img width="182" alt="image" src="https://github.com/woocommerce/woocommerce/assets/73110514/8de5e448-a511-4a1d-96f0-cf6efaeb1230">

Reply:
No worries, I think you're right. Not perfect, but I just added the notice in https://github.com/woocommerce/woocommerce/pull/40505/commits/1cf7a88e258aa7b64756652b37a63f74eefbd85f

![image](https://github.com/woocommerce/woocommerce/assets/444632/6df50e0c-01b8-44d3-b774-948cb1031838)